### DISCUSSION TITLE
New storage inconsistency

### DISCUSSION THREAD

**Discussion by tonytoth** [GrapesJS/grapesjs#4666] [Q&A]
@artf I don't understant what's going on with this new breaking changes.
https://github.com/artf/grapesjs/releases/tag/v0.19.4
I mean the old structure was ```components, styles, assets``` and the new one is with ```pages, styles, assets```.
Ok, I get that but when I load pages I have something like this: 
```
{
    pages: {
        component: [{ some component }]
    }
}
```
and when it comes to save pages the structure is actually different:
```
{
    pages: [
        {
            frames: [{
                component: [{}]
            }]
        }
    ]
}
```
I don't really understand this:
```
The main breaking change in this release, is the https://github.com/artf/grapesjs/pull/4223 in order to bring consistency across various storages (custom and built-in).
```
I mean what is that frame ? Do I need it ? Can I disabled it ?
How can I migrate my database to use this new version of Grapes when I don't know whats going on.

**Comment by artf** [comment]
Yeah, the final projectData JSON contains frames inside pages (you can find more about frames [here](https://github.com/artf/grapesjs/pull/2524)). By loading a JSON containing `pages.component` the editor simply transforms that to the actual format with frames to abstract the need of specifying more stuff in your initial JSON, but the export will always be with pages and frames.
